net/rpmsg/rpmsg_sockif.c

```diff
@@ -505,8 +502,25 @@ static void rpmsg_socket_ns_bind(FAR struct rpmsg_device *rdev,
   strcpy(new->rpaddr.rp_cpu, rpmsg_get_cpuname(rdev));
   strcpy(new->rpaddr.rp_name, name);
 
+  rpmsg_socket_lock(&server->recvlock);
+
+  for (tmp = server; tmp->next; tmp = tmp->next)
+    {
+      if (++cnt >= server->backlog)
+        {
+          /* Reject the connection */
+
+          rpmsg_destroy_ept(&new->ept);
+          rpmsg_socket_free(new);
+          rpmsg_socket_lock(&server->recvlock);
```

Comment:
```suggestion
          rpmsg_socket_unlock(&server->recvlock);
```
Shouldn't you unlock here instead?

Reply:
Yes, you are right !